tests/unit/synapseclient/extensions/unit_test_create_json_schema.py

```diff
@@ -0,0 +1,1310 @@
+"""
+This contains unit test for the crate_json_schema function, and its helper classes and functions.
+The helper classes tested are JSONSchema, Node, GraphTraversalState,
+"""
+
+import json
+import logging
+import os
+import uuid
+from shutil import rmtree
+from typing import Any, Optional
+from unittest.mock import Mock
+
+import pytest
+from jsonschema import Draft7Validator
+from jsonschema.exceptions import ValidationError
+
+from synapseclient.extensions.curator.schema_generation import (
+    DataModelGraphExplorer,
+    GraphTraversalState,
+    JSONSchema,
+    JSONSchemaFormat,
+    JSONSchemaType,
+    Node2,
+    _create_array_property,
+    _create_enum_array_property,
+    _create_enum_property,
+    _create_simple_property,
+    _get_validation_rule_based_fields,
+    _set_conditional_dependencies,
+    _set_property,
+    _set_type_specific_keywords,
+    _write_data_model,
+    create_json_schema,
+)
+
+# pylint: disable=protected-access
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-positional-arguments
+
+
+# Test data paths
+TEST_DATA_BASE_PATH = "tests/unit/synapseclient/extensions"
+SCHEMA_FILES_DIR = f"{TEST_DATA_BASE_PATH}/schema_files"
+EXPECTED_SCHEMAS_DIR = f"{SCHEMA_FILES_DIR}/expected_jsonschemas"
+JSON_INSTANCES_DIR = f"{SCHEMA_FILES_DIR}/json_instances"
+
+# Schema file patterns
+EXPECTED_SCHEMA_PATTERN = "{datatype}.schema.json"
+EXPECTED_DISPLAY_NAMES_SCHEMA_PATTERN = "{datatype}.display_names_schema.json"
+TEST_SCHEMA_PATTERN = "test.{datatype}.schema.json"
+TEST_DISPLAY_NAMES_SCHEMA_PATTERN = "test.{datatype}.display_names_schema.json"
+
+
+# Helper functions for path construction
```

Comment:
The original schematic has file paths hard-coded all over all the place in the test. Please see below as some examples: 
* https://github.com/Sage-Bionetworks/schematic/blob/develop/tests/unit/test_create_json_schema.py#L665
* https://github.com/Sage-Bionetworks/schematic/blob/develop/tests/unit/test_create_json_schema.py#L697
All these hard-coded file paths make maintenance difficult when the file locations change. I changed it so that they are all organized using these helper functions.